### PR TITLE
Add typescript-enum-array plugin

### PR DESCRIPTION
**Pull request by marocchino**
🚨 **IMPORTANT: Please do not create a Pull Request without creating an issue first.**

*Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request.*

## Description

Add new plugin that generates const arrays

Related #6258 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Screenshots/Sandbox (if appropriate/relevant):

Adding links to sandbox or providing screenshots can help us understand more about this PR and take action on it as appropriate

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] I have generate own project with this config:

```yaml
overwrite: true
schema: 'schema.graphql'
documents:
  - 'queries/*.graphql'
generates:
  generated-graphql.ts:
    plugins:
      - typescript:
          enumsAsTypes: true
  generated-const.ts:
    plugins:
      - typescript-enum-array:
          importFrom: './generated-graphql'

```

**Test Environment**:
- OS: MacOS
- `@graphql-codegen/...`: 
- NodeJS: v16.4.1

## Checklist:

- [x] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by dotansimha** [comment]
@marocchino thank you, can you please create a changeset file? (Run `yarn changeset` and it will guide you)

File: packages/plugins/typescript/enum-array/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "@graphql-codegen/typescript-enum-array",
+  "version": "1.0.0",
```
Comment (marocchino) [inline packages/plugins/typescript/enum-array/package.json:3]: Should I down this version?

Comment (dotansimha) [reply packages/plugins/typescript/enum-array/package.json]: Yeah, start with `0.0.0` so next major will be `1.0.0`. Thanks!

**Comment by marocchino** [comment]
I wrote some test for it. Please let me know if there is anything I need to do before merge. ✋🏽

**Review by dotansimha — APPROVED** [review]
This is great! Thank you so much :)

**Pull request merged**